### DISCUSSION TITLE
Persistence Module

### DISCUSSION THREAD

**Discussion by meatballs** [anvilistas/tabulator#147] [Q&A]
I often use the persistence module from Anvil Extras along with the `use_model` option in tabulator. It works well and means I can add properties to my model class so that I get 'computed' columns available to the tabulator.

However, the only option I have is to create a list of model instances as the tabulator data. That's fine for small numbers of rows, but it gets slow when there are much larger numbers involved.

For performance, using a client view of an app table works much better - but then I lose the ability to have those extra computed columns. I'm limited to the columns in the underlying table.

Is there any way to have the best of both? Can I somehow pass each row to a persistence class constructor but lazily rather than as a list?

**Comment by s-cork** [comment]
yes - i can see laziness being highly desirable!

One of the easy wins with search iterators is that there's a built in method for sorting.
So when the user clicks a header to sort by that header, we can easily just request a search iterator with an order_by parameter.

So any lazyness api would also need to provide the ability to speak ordering.

**Reply by meatballs** [reply]
I was hoping to use a tabulator mutator but I hadn't realised those are per column only
